fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/provider/impl/S3Provider.java

```diff
@@ -330,25 +331,22 @@ public void close() throws Exception {
 
     @Override
     public void writeResources(String mediaType, List<ReadResultDTO> dtos) throws Exception {
-        long cosMultiPartMinSize = ConfigurationFactory.getInstance().getCoreCosPartUploadTriggerSize();
-        for (ReadResultDTO dto : dtos) {
-            switch (mediaType) {
-            case FHIRMediaType.APPLICATION_PARQUET:
-                if (chunkData.getPageNum() > chunkData.getLastPageNum() || chunkData.getBufferStream().size() > cosMultiPartMinSize || chunkData.isFinishCurrentUpload()) {
-                    pushFhirParquetToCos(dto.getResources());
-                    chunkData.setLastWritePageNum(chunkData.getPageNum());
-                }
-                break;
-            case FHIRMediaType.APPLICATION_NDJSON:
-            default:
-                if (chunkData.getPageNum() > chunkData.getLastPageNum() || chunkData.getBufferStream().size() > cosMultiPartMinSize || chunkData.isFinishCurrentUpload()) {
-                    // TODO try PipedOutputStream -> PipedInputStream instead?
-                    pushFhirJsonsToCos(new ByteArrayInputStream(chunkData.getBufferStream().toByteArray()), chunkData.getBufferStream().size());
-                    chunkData.setLastWritePageNum(chunkData.getPageNum());
-                }
+        switch (mediaType) {
+        case FHIRMediaType.APPLICATION_PARQUET:
+            pushFhirParquetToCos(dtos.stream()
+                    .flatMap(dto -> dto.getResources().stream())
+                    .collect(Collectors.toList()));
+            break;
+        case FHIRMediaType.APPLICATION_NDJSON:
+        default:
+            // Only if we're greater than zero, otherwise there is nothing to upload.
+            if (chunkData.getBufferStream().size() > 0) {
```

Comment:
you forgot about OperationOutcomes... we should leave the check here... OperationOutcomes should use this going forward in the future.

Reply:
https://github.com/IBM/FHIR/issues/2046